lib/src/algorithms/rsa.dart

```diff
@@ -253,8 +246,7 @@ class RSAKeyParser {
   }
 }
 
-Uint8List _hexToBytes(String encoded) => (Uint8List.fromList(List.generate(
-        encoded.length, (i) => i % 2 == 0 ? encoded.substring(i, i + 2) : null)
-    .where((b) => b != null)
-    .map((b) => int.parse(b, radix: 16))
-    .toList()));
+Uint8List _hexToBytes(String encoded) => (Uint8List.fromList(
```

Comment:
This algorithm didn't work for odd-length strings before, but introduced `null` into the pipeline. Refactoring this into a version that explicitly uses the even-length strings removes the need for nulls and doesn't clutter the type with unnecessary `?`.

Reply:
Sure, we can assert for the right length despite nulling the filtering, makes sense.